bin/prepare_cronet.dart

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// Contains the nessesary setup code only. Not meant to be exposed.
+
+import 'dart:io' show Directory, File, HttpClient, Process;
+
+import 'package:archive/archive.dart';
+import 'package:archive/archive_io.dart';
+import 'package:cronet/src/third_party/ffigen/find_resource.dart';
+import 'package:cronet/src/constants.dart';
+
+import 'package:cli_util/cli_logging.dart' show Ansi, Logger;
+
+/// Builds the `wrapper` shared library for linux.
+bool buildWrapperLinux() {

```

Comment:
Ok. Used gcc because by in linux, we're building the wrapper locally every time. Unlike on windows, where we download the wrapper. As, cmake may not be available in distros by default, I used gcc. Should we move to providing wrapper for linux too?

Reply:
Lets try providing both the wrapper and cronet for now.

Then we can move the buildscripts for both the wrapper and cronet into tool/ and let bin/setup.dart only do downloading.

Reply:
Added `cronet:setup build` and, it can be invoked with `tool/build_wrapper.dart`. For cronet, I'll add the build script under tool only.